tests/regression/lp-1644439/task.yaml

```diff
@@ -0,0 +1,18 @@
+summary: Regression test for https://bugs.launchpad.net/snap-confine/+bug/1644439
+details: |
+    snap-confine uses privately-shared /run/snapd/ns to store bind-mounted
+    mount namespaces of each snap. In the case that snap-confine is invoked
+    from the mount namespace it typically constructs that directory is not
+    available as it is only visible in the main, outer namespace. In order to
+    operate in such an environment snap-confine must first re-associate its own
+    process with another namespace in which the /run/snapd/ns directory is
+    visible. The most obvious candidate is pid one, which definitely doesn't
+    run in a snap-specific namespace, has a predictable PID and is long lived.
```

Comment:
This is such a clear explanation of the rationale. Thanks! Can we please have this copied over above the function sc_reassociate_with_pid1_mount_ns in ns-support.c as well?